File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/conveniocategoria/dto/ConvenioCategoriaGridDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.conveniocategoria.dto;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Getter
@Setter
public class ConvenioCategoriaGridDTO implements GridDTO {
    private UUID id;
    private String convenioNome;
    private String nome;
    private String codigoAnsPlano;
    private Boolean ativo;
    private LocalDateTime createdAt;
    private Boolean deleted;
}

File: src/frontend/src/app/components/atendimento/convenio-categoria/model/convenio-categoria-grid-dto.ts
export class ConvenioCategoriaGridDTO {
  id: string;
  convenioNome?: string;
  nome: string;
  codigoAnsPlano?: string;
  ativo?: boolean;
  deleted?: boolean;

  constructor(
    id: string,
    nome: string,
    convenioNome?: string,
    codigoAnsPlano?: string,
    ativo?: boolean,
    deleted?: boolean
  ) {
    this.id = id;
    this.nome = nome;
    this.convenioNome = convenioNome;
    this.codigoAnsPlano = codigoAnsPlano;
    this.ativo = ativo;
    this.deleted = deleted;
  }
}
